### PR TITLE
Cloud datastore links

### PR DESCRIPTION
**Pull request by MaksYermak**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/links/datastore.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        task_instance.xcom_push(
+            context=context,
+            key=CloudDatastoreImportExportLink.key,
+            value={
+                "project_id": task_instance.project_id,
+            },
+        )
+
+
+class CloudDatastoreExportEntitiesLink(BaseGoogleLink):
```
Comment (lwyszomi) [inline airflow/providers/google/cloud/links/datastore.py:55]: Can we use this link class: https://github.com/apache/airflow/blob/main/airflow/providers/google/common/links/storage.py

Comment (MaksYermak) [reply airflow/providers/google/cloud/links/datastore.py:55]: @lwyszomi we can't, this link to specific folder inside the bucket

Comment (lwyszomi) [reply airflow/providers/google/cloud/links/datastore.py:55]: but you can  specify the `uri` parameter that can be passed to the `StorageLink`, so you can execute something like this:
```python
StorageLink.persist(uri=f"{task_instance.bucket}/{output_url.split('/')[3]}")
```

Comment (MaksYermak) [reply airflow/providers/google/cloud/links/datastore.py]: @lwyszomi I have updated the code

**Review by lwyszomi — APPROVED** [review]

**Pull request closed without merging**